registry/server/common/services/assets/AssetsDiscovery.ts

```diff
@@ -3,98 +3,147 @@ import urljoin from 'url-join';
 import deepEqual from 'deep-equal';
 
 import knex from '../../../db';
-import manifestProcessor from './assetsManifestProcessor';
+import manifestProcessor, { ManifestData } from './assetsManifestProcessor';
 import AssetsDiscoveryWhiteLists from './AssetsDiscoveryWhiteLists';
 import { getLogger } from '../../../util/logger';
 import { parseJSON } from '../json';
 import { axiosErrorTransformer } from '../../../util/axiosErrorTransformer';
+import { startBackgroundTransaction } from 'newrelic';
+
+type AssetsDiscoveryEntity = {
+    [key: string]: any;
+    assetsDiscoveryUrl: string;
+    spaBundle: string;
+    cssBundle: string;
+    dependencies: string;
+    assetsDiscoveryUpdatedAt: number;
+    props?: object | null;
+};
+
+type DbAssetsOnly = Pick<AssetsDiscoveryEntity, 'spaBundle' | 'cssBundle' | 'dependencies'>;
 
 export default class AssetsDiscovery {
-    private tableName: string;
-    private tableId: string;
+    private tableName: keyof typeof AssetsDiscoveryWhiteLists;
+
+    private tableId: 'name';
+
     private timerId?: NodeJS.Timeout;
 
     private intervalSeconds: number;
 
-    constructor(tableName: string, { tableId = 'name', intervalSeconds = 5 } = {}) {
+    constructor(
+        tableName: keyof typeof AssetsDiscoveryWhiteLists,
+        { tableId = 'name' as const, intervalSeconds = 5 } = {},
+    ) {
         this.tableName = tableName;
         this.tableId = tableId;
         this.intervalSeconds = intervalSeconds;
     }
 
     start(delay: number = 1000) {
-        this.timerId = setInterval(
-            () =>
-                this.iteration().catch((err) => {
+        const runLoop = () => {
+            this.timerId = setTimeout(async () => {
+                await this.iteration().catch((err) => {
                     getLogger().error(err, 'Error during refresh of the assets info:');
-                }),
-            delay,
-        );
+                });
+                runLoop();
+            }, delay);
+        };
+
+        runLoop();
     }
 
     stop() {
-        clearInterval(this.timerId!);
+        clearTimeout(this.timerId!);
+    }
+
+    private async iteration() {
+        return startBackgroundTransaction(this.tableName, AssetsDiscovery.constructor.name, async () => {
+            const now = Math.floor(Date.now() / 1000);
```

Comment:
I still beleive that enire logic should be changed in order to take in a count
all the previous responses time/timeouts in updatedTime parameter.

Reply:
Noted